src/models/ratings/ratings.service.ts

```diff
@@ -0,0 +1,443 @@
+									Math.floor(whiteAdjustedRating),
+									Math.floor(blackAdjustedRating),
+									Math.round((whiteAdjustedRating2 - whiteAdjustedRating) * 10),
+									Math.round((blackAdjustedRating2 - blackAdjustedRating) * 10),
+									game.id,
+								);
+							}
+						} else {
+							if (
+								quickResult === undefined &&
+								game.date > RECENT_LIMIT
+							) {
+								updating = false;
+							} else {
+								lastUsedGame = game.id;
+								await updateGame(Math.floor(whiteAdjustedRating), Math.floor(blackAdjustedRating), -2000, -2000, game.id);
+							}
+						}
+					}
+				} else {
+					if (updating) {
+						lastUsedGame = game.id;
+					}
+					await updateGame(Math.floor(whiteAdjustedRating), Math.floor(blackAdjustedRating), -2000, -2000, game.id);
+				}
+			}
+
+			async function updatePlayer(rating: number, boost: number, ratedGames, maxRating, ratingAge, fatigueObject, id, fatigueRating) {
+				// update player query
+				const updatePlayerQuery = `UPDATE players SET
+				rating=?, boost=?, ratedgames=?, maxrating=?, ratingage=?, fatigue=?, fatiguerating=? where id=?;`;
+				try {
+					await playerRunner.manager.query(updatePlayerQuery, [
+						rating,
+						boost,
+						ratedGames,
+						maxRating,
+						ratingAge,
+						JSON.stringify(fatigueObject),
+						fatigueRating,
+						id,
+					]);
+				} catch (error) {
+					has_error = true;
+					this.logger.error('Error updating player. ', error);
+					throw new Error(error);
+				}
+			}
+			this.logger.debug('Updating players');
+			for (let i = 0; i < playersArray.length; i++) {
+				if (playersArray[i].changed) {
+					await updatePlayer(
+						playersArray[i].rating,
+						playersArray[i].boost,
+						playersArray[i].ratedgames,
+						playersArray[i].maxrating,
+						playersArray[i].ratingage,
+						JSON.stringify(playersArray[i].fatigue),
+						this.adjustedRating(playersArray[i], Date.now(), PARTICIPATION_CUTOFF, RATING_RETENTION, MAX_DROP, PARTICIPATION_LIMIT),
+						playersArray[i].id,
+					);
+				}
+				// always update fatiguerating
+				const updateFatigueRatingQuery = `UPDATE players SET fatiguerating=? where id=?;`;
+				await playerRunner.manager.query(updateFatigueRatingQuery, [this.adjustedRating(playersArray[i], Date.now(), PARTICIPATION_CUTOFF, RATING_RETENTION, MAX_DROP, PARTICIPATION_LIMIT), playersArray[i].id]);
+			}
+			await playerRunner.commitTransaction();
+			await gameRunner.commitTransaction();
+		} catch (error) {
+			console.error(error);
+			this.logger.error("Error processing player and game ratings, rolling back transaction. ", error);
+			playerRunner.rollbackTransaction();
+			gameRunner.rollbackTransaction();
+		} finally {
+			playerRunner.release();
+			gameRunner.release();
+			if (!has_error) {
+				this.logger.debug('Finished updating ratings');
+				await writeFile('previous.txt', lastUsedGame + ' ' + 'hash');
+			}
+		}
+	}
+
+	// returns the adjusted rating of a player
+	public adjustPlayer(
```

Comment:
Note that this method and updateFatigue are mutations; they change the `player` object that is passed in. The description suggests otherwise. That may be fine though?

Reply:
Updated the comments to explain what its doing it better.